support-frontend/assets/components/marketingConsent/marketingConsent.jsx

```diff
@@ -13,19 +13,26 @@ import { logException } from 'helpers/logger';
 import Button from 'components/button/button';
 import NonInteractiveButton from 'components/button/nonInteractiveButton';
 import 'components/marketingConsent/marketingConsent.scss';
+import type { ThankYouPageMarketingComponentTestVariants } from 'helpers/abTests/abtestDefinitions';
 
 // ----- Types ----- //
 
 type ButtonPropTypes = {|
   confirmOptIn: ?boolean,
   email: string,
   csrf: CsrfState,
-  onClick: (?string, CsrfState) => void,
+  onClick: (?string, CsrfState, ?ThankYouPageMarketingComponentTestVariants) => void,
   requestPending: boolean,
+  marketingComponentVariant?: ThankYouPageMarketingComponentTestVariants,
 |};
 
 type PropTypes = {|
-  ...ButtonPropTypes,
```

Comment:
For some reason, doing the object spread here was causing eslint errors - I don't think it likes it when it comes to optional props

Reply:
that's annoying. At least we can revert this part later

Reply:
👍